pipeline-modules/deployment-service/gcp/run-bash-script.yaml

```diff
@@ -0,0 +1,89 @@
+provisioner: gcp
+name: run-bash-script
+version: 1
+revision: 1
+displayName: Execute a bash script
+description: Execute a bash script from a file
+target: "deployment-template"
+keywords:
+  - bash
+  - linux
+author: CloudCover
+meta:
+  inputArtifactType:
+    - GitCode
+
+inputs:
+  properties:
+    bash_script:
+      title: Bash script
+      description: This the filename of the bash script to run
+      type: string
+    script_parameters:
+      title: Parameters to pass to the bash script
+      type: array
+      default: []
+      items:
+        type: string
+    repo_dir:
+      title: Repo Dir
+      description: Repo Dir
+      type: string
+    job_type:
+      title: Job Type
+      description: This is to deploy or undeploy application
+      type: string
+      default: deploy
+    pipeline_summary_var:
+      title: Pipeline Summary
+      description: This will create pipeline summary as per user request
+      type: array
+      default: []
+      items:
+        type: object
+        required: [Command, Name]
+        properties:
+          Command:
+            title: Command
+            type: string
+          Name:
+            title: Name
+            type: string
+  required:
+    - bash_script
+  internal:
+    - job_type
+    - pipeline_summary_var
+    - repo_dir
+template: |
+  {% set workspace = '/workspace' %}   {# has to be a persistent volume #}
+  {% set pipeline_env_file = workspace|add:'/.env' %}
+  {% set pipeline_secret_env_file = workspace|add:'/.secret.env' %}
+  {% set source_code_path = workspace|add:'/source_code' %}
+
+  steps:
+  - id: 'Execute bash script'
+    name: 'gcr.io/cloud-builders/gcloud'
+    entrypoint: 'bash'
+    args:
+    - '-c'
+    - |
+      export CODEPIPES_DEPLOY_ACTION={{ job_type }}
```

Comment:
Similar to Azure case I think you could define an env: section in this particular step.